src/qibo/validation/gate_set_tomography.py

```diff
@@ -0,0 +1,2089 @@
+
+            elif j == 13:  # Z basis  # X basis
+                circ2 = Circuit(nqubits, density_matrix=True)
+                new_circ = circ + circ2
+                if noise_model is not None and backend.name != "qibolab":
+                    new_circ = noise_model.apply(new_circ)
+                new_circ.add(gates.M(0, basis=gates.Z))
+                new_circ.add(gates.M(1, basis=gates.X))
+
+                expectation_val = GST_measure_2qb(new_circ, k, j)
+
+            elif j == 14:  # Z basis  # Y basis
+                circ2 = Circuit(nqubits, density_matrix=True)
+                new_circ = circ + circ2
+                if noise_model is not None and backend.name != "qibolab":
+                    new_circ = noise_model.apply(new_circ)
+                new_circ.add(gates.M(0, basis=gates.Z))
+                new_circ.add(gates.M(1, basis=gates.Y))
+
+                expectation_val = GST_measure_2qb(new_circ, k, j)
+
+            elif j == 15:  # Z basis # Z basis
+                circ2 = Circuit(nqubits, density_matrix=True)
+                new_circ = circ + circ2
+                if noise_model is not None and backend.name != "qibolab":
+                    new_circ = noise_model.apply(new_circ)
+                new_circ.add(gates.M(0, basis=gates.Z))
+                new_circ.add(gates.M(1, basis=gates.Z))
+
+                expectation_val = GST_measure_2qb(new_circ, k, j)
+
+            # print(expectation_val)
+            matrix_jk_2qb[j, k] = expectation_val
+
+    matrix_jk_2qb = np.matrix(matrix_jk_2qb)
+
+    return matrix_jk_2qb
+
+
+def GST_1qb_basis_operations(
+    NshotsGST=1e4, noise_model=None, gjk_1qb=None, save_data=None, backend=None
+):
+    """Runs gate set tomography for the 13 basis operations for a single qubit.
+
+        Args:
+        NshotsGST (int, optional): Number of shots used in Gate Set Tomography.
+        noise_model (:class:`qibo.noise.NoiseModel`, optional): Noise model applied
+            to simulate noisy computation.
+        gjk_1qb (numpy.matrix): Matrix with elements Tr(Q_j rho_k) for one qubit with no gate.
+            If None, call GST_1qb without gate.
+        save_data: Flag to save gate set tomography data. If None, skip.
+        backend (:class:`qibo.backends.abstract.Backend`, optional): Calculation engine.
+
+        The 13 basis operations are based on Physical Review Research 3, 033178 (2021).
+    Returns:
+        numpy.matrix: Matrix with elements jk equivalent to either Tr(Q_j rho_k) or Tr(Q_j O_l rho_k).
+    """
+
+    import numpy as np
+
+    from qibo import gates
+    from qibo.backends import GlobalBackend
+    from qibo.models import Circuit
+
+    if backend is None:  # pragma: no cover
+        backend = GlobalBackend()
+
+    if gjk_1qb is None:
+        gjk_1qb = GST_1qb(NshotsGST, noise_model, save_data=None)
+
+    BasisOps_13 = [
+        np.matrix(gates.I(0).matrix(backend=GlobalBackend())),
```

Comment:
Why do you need the ``np.matrix``? Isn't a ``np.array`` enough?

Reply:
I initially used  ```np.matrix(gates.I(0).matrix(backend=GlobalBackend())) ``` and the code managed to run. 

However, there were problems within the resulting basis operations in matrix form when I checked. I found that having the additional ```np.matrix()``` outside of gates.I(0).matrix(backend=GlobalBackend()))``` resolved the matrix errors.

Reply:
But having ``np.matrix`` is complicating the code, for example above you needed to convert back to ``np.arrays``. I would avoid using them and if errors pop out, try understanding why they pop out and correct them.

Reply:
Hmm! If I'm not wrong, I only do the array operation once. Most of the code deals with matrices.

I must correct my previous comment for it is inaccurate.

Previously, I used ```gates.I(0).matrix(backend=GlobalBackend())``` and the code managed to run. However, when I printed the basis operations to check, they were inaccurate. I found that converting adding ```np.matrix()``` outside of ```gates.I(0).matrix(backend=GlobalBackend())``` resolved the problem.

Sorry, the previous comment was wrong. My bad.

Reply:
Can you please move everything to ``np.arrays`` anyway?

Reply:
I'm a little hesitant to do that because checking through 13 4x4 and 241 16x16 matrices is not a pretty task.

But I'm sensing that numpy arrays is the preferred option?

If yes then I'll do that once I think about the helper function for "j" measurement bases.

Reply:
I don't see what is the benefit of using ``np.matrix`` at the moment. 

> I'm a little hesitant to do that because checking through 13 4x4 and 241 16x16 matrices is not a pretty task.
That's also one of the reason why I was saying that is better to have general helper functions that construct what you need, and not repeat hard coded definitions over and over.

Reply:
Changed all ```np.matrix```'s to ```np.array```s.